src/dataplane-protocol/src/batch.rs

```diff
@@ -27,7 +24,7 @@ pub trait BatchRecords: Default + Debug + Encoder + Decoder {
     }
 }
 
-impl BatchRecords for DefaultBatchRecords {}
+impl BatchRecords for Vec<Record> {}
```

Comment:
I would like to create alias for `Vec<Records>` this might be new type in future.